core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -368,6 +368,20 @@ abstract class Chunk[+O] extends Serializable with ChunkPlatform[O] with ChunkRu
   /** Converts this chunk to a scodec-bits BitVector. */
   def toBitVector[B >: O](implicit ev: B =:= Byte): BitVector = toByteVector[B].bits
 
+  def traverse_[F[_]](f: O => F[Unit])(implicit F: Applicative[F]): F[Unit] =
+    if (isEmpty) F.unit
+    else
+      F match {
+        case m: FlatMap[F] =>
+          val s = size
+          def go(i: Int): F[Unit] =
+            // this is assuming our F is tail-recursive. Is that fair assumption?
+            if (i < s) m.flatMap(f(apply(i)))(_ => go(i + 1)) else F.unit
```

Comment:
I don't understand this comment. Do you mean that `F` is stacksafe? In that case why not pattern match for `StackSafeMonad` ?